### PR TITLE
Add Line Item model

### PR DESCRIPTION
**Pull request by danielx0328**
### Goals :dart:
Define the Line Item model used in the Order model.

### Implementation Details :construction:
Added the following properties:
* identifier
* total
* quantity
* itemDescription
* sku
* upc
* category
* attributes

### Testing Details :mag:
Added unit tests for the new model

### CODE REVIEW

File: Source/Order.swift [hunk 1]
```diff
@@ -159,4 +152,93 @@ Represents an order placed by the user to be tracked using `ButtonMerchant.track
...
+        /**
+         The total price of all items bought in a particular line item (required).
+         (e.g. if 3 bananas were purchased for $3.00 each, total would be 900).
+         */
+        let total: Int64
+
+        /**
+         The number of unique units represented by this line item.
+         */
+        let quantity: Int
```
Comment (wessmith) [inline Source/Order.swift]: add `(required).` to be consistent with the other properties

Comment (Jon6193) [reply Source/Order.swift]: Quantity is optional so we should add a ? and default to nil

Comment (danielx0328) [reply Source/Order.swift]: @wessmith quantity is not required

Comment (wessmith) [reply Source/Order.swift]: Was going by the doc

Screen Shot 2019-06-24 at 2 52 29 PM

Comment (wessmith) [reply Source/Order.swift]: Screen Shot 2019-06-24 at 2 54 04 PM

Comment (Jon6193) [reply Source/Order.swift]: I'll update the doc accordingly

File: Source/Order.swift [hunk 2]
```diff
@@ -159,4 +154,81 @@ Represents an order placed by the user to be tracked using `ButtonMerchant.track
...
+        /**
+         The total price of all items bought in a particular line item (required).
+         (e.g. if 3 bananas were purchased for $3.00 each, total would be 900).
+         */
+        let total: Int64
+
+        /**
+         The number of unique units represented by this line item.
+         */
+        public var quantity: Int
```
Comment (wessmith) [inline Source/Order.swift]: Are we going with a default value or excluding this field?

File: Source/Order.swift [hunk 3]
```diff
@@ -159,4 +154,81 @@ Represents an order placed by the user to be tracked using `ButtonMerchant.track
...
+         - Parameters:
+            - identifier: The unique identifier for this line item, within the scope of this order.
+                          This must be unique across all line-items within the order.
+                          We suggest using the SKU or UPC of the product. (required)
+            - total: The total price of all items bought in a particular line item. (required)
+         */
+        @objc public init(identifier: String, total: Int64) {
+            self.identifier = identifier
+            self.total = total
+            self.quantity = 1
```
Comment (wessmith) [inline Source/Order.swift]: @Jon6193 are we also defaulting to 1 on Android?

File: Tests/UnitTests/OrderTests.swift [hunk 4]
```diff
@@ -126,48 +126,108 @@ class OrderTests: XCTestCase {
...
+        XCTAssertEqual(lineItem.total, total)
+        XCTAssertEqual(lineItem.quantity, 1)
+        XCTAssertNil(lineItem.itemDescription)
+        XCTAssertNil(lineItem.sku)
+        XCTAssertNil(lineItem.upc)
+        XCTAssertNil(lineItem.category)
+        XCTAssertNil(lineItem.attributes)
+    }
+
+    func testLineItemInitialization_allProperties() {
```
Comment (wessmith) [inline Tests/UnitTests/OrderTests.swift]: Are we covering serialization elsewhere?

File: Source/Order.swift (hunk 3 above)
Comment (Jon6193) [reply Source/Order.swift]: Yeah we will

File: Tests/UnitTests/OrderTests.swift (hunk 4 above)
Comment (danielx0328) [reply Tests/UnitTests/OrderTests.swift]: When we have a finalized model, then a test for serialization will be added

File: Source/Order.swift (hunk 2 above)
Comment (danielx0328) [reply Source/Order.swift]: default provided on init

File: Source/Order.swift (hunk 3 above)
Comment (Jon6193) [reply Source/Order.swift]: https://github.com/button/button-merchant-android/pull/23

**Pull request merged**